recommender/recommender.py

```diff
@@ -586,7 +587,7 @@ def add_resource(self, data, _suffix=''):  # pylint: disable=unused-argument
         for field in self.resource_content_fields:
             result[field] = strip_and_clean_html_elements(data[field])
 
-        resource_id = stem_url(data['url'])
+        resource_id = stem_url(result['url'])
```

Comment:
@fysheets: What does stem_url do?  I don't see it being a security issue if you get this wrong, but the `_check_redundant_resource` and `_check_removed_resource` checks below might not do what are intended.  May not be a big deal?

Reply:
Looking at the code:

```def stem_url(url):
    """
    Get the base form of url.
    This is not designed for security, just to check common errors/use-cases.
    """
    parsed_url = urlparse(unquote_plus(url))
    return urlunparse(parsed_url._replace(fragment=''))
```

Looks like it does some cleaning as well, but calls out that it's not for security.

Reply:
Ok.  We don't have to touch this then.